httplog/httplog.go

```diff
@@ -32,6 +33,108 @@ func (m Mode) String() string {
 	return string(m)
 }
 
+type ModuleMode struct {
+	Name string
+	Mode *Mode
+}
+
+func NewModuleMode(s string) (ModuleMode, error) {
+	n, m, ok := strings.Cut(s, ":")
+	if !ok {
+		n = ""
+		m = s
+	}
+
+	var mode Mode
+
+	switch Mode(m) {
+	case None:
+		mode = None
+	case ShortURL:
+		mode = ShortURL
+	case URL:
+		mode = URL
+	case Headers:
+		mode = Headers
+	case Body:
+		mode = Body
+	case Errors:
+		mode = Errors
+	default:
+		return ModuleMode{}, fmt.Errorf("invalid mode %q", m)
+	}
+
+	return ModuleMode{Name: n, Mode: &mode}, nil
+}
+
+func (m ModuleMode) String() string {
+	if m.Name == "" {
+		return m.Mode.String()
+	}
+
+	return m.Name + ":" + m.Mode.String()
+}
+
+type Config []ModuleMode
+
+func (c *Config) Append(s string) error {
+	m, err := NewModuleMode(s)
+	if err != nil {
+		return err
+	}
+	*c = append(*c, m)
+
+	return nil
+}
+
+func (c *Config) Replace(s []string) error {
+	c = nil
+
+	for _, m := range s {
+		n, err := NewModuleMode(m)
+		if err != nil {
+			return err
+		}
+		*c = append(*c, n)
+	}
+
+	return nil
+}
+
+func (c Config) GetSlice() []string {
+	result := make([]string, len(c))
+
+	for i, s := range c {
+		result[i] = s.String()
+	}
+
+	return result
+}
+
+func (c Config) String() string {
+	return strings.Join(c.GetSlice(), ",")
+}
+
+func (c Config) Set(s string) error {
+	m, err := NewModuleMode(s)
+	if err != nil {
+		return err
+	}
+
+	for i := range c {
+		if c[i].Name == m.Name {
+			*c[i].Mode = *m.Mode
+			return nil
+		}
+	}
```

Comment:
If there is no fallback to "" we can have anyflag implementation and avoid the boilerplate.

I'm thinking that you could also have a wrapper type around anyflag type that would overwrite Set and Replace adding post processing i.e. if we have default specified and the flag is unset use the default.